sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -0,0 +1,416 @@
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(6, eventHubName ?? String.Empty, partitionId ?? String.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that the receiving of events has completed.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub being received from.</param>
+        /// <param name="partitionId">The identifier of the partition events are being received from.</param>
+        /// <param name="eventCount">The number of events that were received in the batch.</param>
+        ///
+        [Event(7, Level = EventLevel.Informational, Message = "Completed receiving events for Event Hub: {0} (Partition Id: '{1}', Event Count: '{2}').")]
+        public void EventReceiveComplete(string eventHubName,
+                                         string partitionId,
+                                         int eventCount)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(7, eventHubName ?? String.Empty, partitionId ?? String.Empty, eventCount);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while receiving events.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub being received from.</param>
+        /// <param name="partitionId">The identifier of the partition events are being received from.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        ///
+        [Event(8, Level = EventLevel.Error, Message = "An exception occurred while receiving events for Event Hub: {0} (Partition Id: '{1}'). Error Message: '{2}'")]
+        public void EventReceiveError(string eventHubName,
+                                      string partitionId,
+                                      string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(8, eventHubName ?? String.Empty, partitionId ?? String.Empty, errorMessage ?? String.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that a client is closing, which may corespond to an <see cref="EventHubClient" />,
+        ///   <see cref="EventHubProducer" />, or <see cref="EventHubConsumer" />.
+        /// </summary>
+        ///
+        /// <param name="clientType">The type of client being closed.</param>
+        /// <param name="eventHubName">The name of the Event Hub associated with the client.</param>
+        /// <param name="clientId">An identifier to associate with the client.</param>
+        ///
+        [Event(9, Level = EventLevel.Verbose, Message = "Closing a {0} client (EventHub '{1}'; Identifier '{2}').")]
+        public void ClientCloseStart(Type clientType,
+                                     string eventHubName,
+                                     string clientId)
```

Comment:
Do we have something like a 'Client Id' in Track Two?

Reply:
Not in the same way as track one, but we need a way to indicate that the same thing started/stopped.   I'm thinking of using the hash code for that, otherwise, we'll generate something and and assign it.